File: Arrays & Hashing/LongestConsecutiveSequence.java
import java.util.*;
public class LongestConsecutiveSequence{
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int i = 0; i<nums.length; i++){
            set.add(nums[i]);
        }

        int length = 0;
        int answer = 0;

        for (int i = 0; i<nums.length; i++){
            // if its a start of a sequence
            if (!set.contains(nums[i] - 1)){
                length = 1;
                while(set.contains(nums[i] + length)){
                    length++;
                }
                answer = Math.max(answer, length);
            }
        }




        return answer;
    }
}

File: Arrays & Hashing/LongestConsecutiveSequence.cpp
#include <bits/stdc++.h>
using namespace std;

int longestConsecutive(vector<int> &nums)
{
    unordered_set<int> set;

    for (auto x : nums)
    {
        set.insert(x);
    }

    int length = 0;
    int ans = 0;

    for (int i = 0; i < nums.size(); i++)
    {

        if (set.find(nums[i] - 1) == set.end())
        {
            length = 1;
            while (set.find(nums[i] + length) != set.end())
            {
                length++;
            }
            ans = max(ans, length);
        }
    }

    return max(length, ans);
}